napari/_qt/qt_main_window.py

```diff
@@ -1149,22 +1154,38 @@ def _screenshot(self, flash=True, canvas_only=False) -> 'QImage':
         from .utils import add_flash_animation
 
         if canvas_only:
-            img = self._qt_viewer.canvas.native.grabFramebuffer()
-            if flash:
-                add_flash_animation(self._qt_viewer._canvas_overlay)
+            canvas = self._qt_viewer.canvas
+            if size is not None:
+                if len(size) != 2:
+                    raise ValueError(
+                        f'screenshot size must be 2 values, got {len(size)}'
+                    )
+                prev_size = canvas.size
+                canvas.size = size[::-1]  # invert x ad y for vispy
```

Comment:
```suggestion
                canvas.size = tuple([dim * size for dim in canvas.size])
```
Obviously with this change one would rename `size` to `scale`, but this works and interestingly doesn't pixel double. And it permits something like:
`screen = viewer.screenshot(size=1//viewer.camera.zoom)`
which works quite nice!

Reply:
Ah, sorry, saw this just now :)

Reply:
NP, this is the first time I've managed to figure out how to use this sort of comments, so rough around the edges.

The more I thin about it, the more I think that if the resizing is not related to the layer size (the data size)—which the user should know and could thus enter, like I was doing—then *using a scale factor* rather than giving a new size is more intuitive. Especially, since the output, a screenshot, is likely to be used *on screen* in a presentation, blog, article, tweet, webpage, etc. So then it's intuitive that I want the image on screen 2x as big or 4x as big. A specific size would be nice if there was a reliable way to eliminate the stray canvas beyond the image data.